conu/utils/filesystem.py

```diff
@@ -222,3 +222,48 @@ def __unicode__(self):
     def __str__(self):
         # we could be possible initialize here, but... it's tricky
         return str(self.path)
+
+
+class Volume(object):
+    """
+    The representation of container volume.
+    """
+
+    def __init__(self, target, source=None, mode=None):
+        if source and not isinstance(source, Directory):
+            self.source = Directory(path=source)
+        else:
+            self.source = source
+        self.target = target
+        self.mode = mode
+
+    def __str__(self):
+        """
+        Cmd option representing the volume
+        :return:
+        """
+        result = self.target
+        if self.source:
+            result = "{}:{}".format(self.source.path, result)
```

Comment:
what if `self.source` is a string?

Reply:
It is converted to `Directory` in `__init__`:
```
        if source and not isinstance(source, Directory):
            self.source = Directory(path=source)
```

Reply:
/facepalm

(I missed that)